lib/utils.js

```diff
@@ -606,6 +606,77 @@ function decorateWithReadConcern(command, coll) {
   }
 }
 
+const emitProcessWarning = msg => process.emitWarning(msg, 'DeprecationWarning');
+const emitConsoleWarning = msg => console.error(msg);
+const emitDeprecationWarning = process.emitWarning ? emitProcessWarning : emitConsoleWarning;
+
+/**
+ * Default message handler for generating deprecation warnings.
+ *
+ * @param {string} [name] function name
+ * @param {string} [option] option name
+ * @return {string} warning message
+ * @ignore
+ * @api private
+ */
+function defaultMsgHandler(name, option) {
+  return name + ' option [' + option + '] is deprecated and will be removed in a later version.';
+}
+
+/**
+ * Deprecates a given function's options.
+ *
+ * @param {object} [config] configuration for deprecation
+ * @param {string} [config.name] function name
+ * @param {Array} [config.deprecatedOptions] options to deprecate
+ * @param {number} [config.optionsIndex] index of options object in function arguments array
+ * @param {function} [config.msgHandler] optional custom message handler to generate warnings
+ * @param {function} [fn] the target function of deprecation
+ * @return {function} modified function that warns once per deprecated option, and executes original function
+ * @ignore
+ * @api private
+ */
+function deprecateOptions(config, fn) {
```

Comment:
You might want to check here if they are running with no deprecations, and just short-circuit the function.